base/range.jl

```diff
@@ -264,6 +264,11 @@ RangeStepStyle(::Type{<:AbstractRange{<:Integer}}) = RangeStepRegular()
 
 convert(::Type{T}, r::AbstractRange) where {T<:AbstractRange} = r isa T ? r : T(r)::T
 
+# there is no complex range
+AbstractRange{T}(r::AbstractRange) where T<:Real = T(first(r)):T(step(r)):T(last(r))
+AbstractArray{T,1}(r::AbstractRange) where T<:Real = AbstractRange{T}(r)
```

Comment:
```suggestion
AbstractVector{T}(r::AbstractRange) where T<:Real = AbstractRange{T}(r)
```

Reply:
see the fix https://github.com/JuliaLang/julia/pull/48894/commits/4e9740d38986b34f747a9694e24a15ab1754e573
I guess the name `AbstractVector` is not accessible there.